### ISSUE TITLE
reseed the rng for every mission when generating demos

### ISSUE THREAD

**Issue by rizar** [mila-udem/babyai#23]
It is currently very tricky to debug occasional bot failures because the environment needs to be reset a  number of times. Perhaps we could reseed the random number generator for every mission. The only pitfall  I can think of is that we should still make sure that the resulting data never overlaps with validation or test sets.

**Comment by maximecb** [comment]
We technically don't thoroughly check that the data doesn't overlap right now. We set a given seed and generate missions, set a different seed and generate a different set of missions in sequence. Would it be acceptable to just say, training data is seeds in the range 1 to 1 million, testing is after 1 million?

Just to be rigorous, you could generate a set of demos that way and I could write a piece of code that checks if, for each demo, the same mission string + initial env config is present in the training set. Probably better than just assuming we're ok.

**Comment by rizar** [comment]
Given how complex the levels are, I believe it would be sufficient to
always use different seeds for training and testing. You could verify this
with a script if you want. I am going through the code right now and
checking our usage of random seeds. Most likely, your suggestion is OK,
although I would go for a bit more future-proof pool of training missions,
reserving seeds from 1 to 10^9 for training.

**Comment by maximecb** [comment]
There is a call to `evaluate()` in `train_intelligent_expert.py`: https://github.com/mila-udem/babyai/blob/master/scripts/train_intelligent_expert.py#L328

As long as this method automatically starts from seeds seeds >= 10^9 we should be ok.

**Comment by rizar** [comment]
The line you are referring to calls
`babyai.imitation.ImitationLearning.validate()`, which in turns calls
`evaluate()` with `args.val_seed`, which should be equal to `int(1e9)`,
cause this is its default value.
